crates/topos-test-sdk/src/integration/mod.rs

```diff
@@ -0,0 +1,154 @@
+use futures::{future::join_all, StreamExt};
+use rand::seq::SliceRandom;
+use std::time::Duration;
+use tokio::time::error::Elapsed;
+use tonic::transport::Uri;
+
+use topos_core::{
+    api::grpc::{
+        shared::v1::checkpoints::TargetCheckpoint,
+        tce::v1::{
+            api_service_client::ApiServiceClient,
+            console_service_client::ConsoleServiceClient,
+            watch_certificates_request::OpenStream,
+            watch_certificates_response::{CertificatePushed, Event},
+            StatusRequest, SubmitCertificateRequest,
+        },
+    },
+    uci::{Certificate, CERTIFICATE_ID_LENGTH, SUBNET_ID_LENGTH},
+};
+use tracing::{debug, info};
+
+pub async fn check_certificate_delivery(
```

Comment:
Given this is fairly involved it would be good to have a few lines explaining what happens. "Picks a random peer and sends it a certificate. All other peers listen for broadcast certs. Three possible outcomes: 1. No errors, returns `Ok`; 2. There were errors, returns a list of all errors encountered; 3. timeout" or something like that?